### PR TITLE
[Issue 185] Branch Convention and Release Workflow ADR

### LINKED ISSUE #185: [ADR]: Branch conventions and release workflow
### Description

This ADR will record the devision for branch naming conventions and release workflow. This may be combined with #112 , but it has been useful internally to treat them as separate tasks.

### Approvers

@acouch 
@widal001 
@daphnegold 
@lucasmbrown-usds (optional) 

### Options

_No response_

### Decision Criteria

_No response_

### Definition of Done

- [ ] The approvers for this decision have been identified (ideally before work on the ADR starts)
- [ ] The ADR is created and stored in `documentation/decisions/adr` with status "Accepted"
- [ ] The ADR has been reviewed and approved by the approvers listed above
- [ ] The ADR satisfies requirements that are outlined in the ADR template
- [ ] Any follow-up tickets have been created (if necessary)

### PR DESCRIPTION
**Pull request by acouch**
## Summary
Fixes #185 

## Changes proposed
Adds ADR described in #185 

## Context for reviewers
This merges #185 and #112 .

### CODE REVIEW

**Review by SammySteiner — APPROVED** [review]
Looks great. Thanks for documenting all of this.
Seems like the google docs are only accessible with a navapbc gmail, so we'd need to copy that info to the repo for it to be publicly accessible.

File: documentation/decisions/2023-08-21-branch-conv-release-workflow.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+### Release workflow and Code Reviews
+
+The following conventions will be adopted:
+
+- Pull Requests (PRs) will be merged directly into `main`. 
+- PRs will be titled `[Issue N] Short description`
+  - PRs with no issue should use `[Fix] Short description`
+- PRs will follow, with small changes, the [Code Change Lifecycle / Contributing
+](https://docs.google.com/document/d/1EyLYuKCv8xjpY26zR8cODl6SEXy9Mx_yAuZ6vS8cPp8/edit?skip_itp2_check=true&pli=1) guidelines and be documented in `CONTRIBUTING.md`
+- Code reviews will follow, with small changes, the [Code Review Guidelines](https://docs.google.com/document/d/1pRtpg1ffLXcJn_yV_g90t1TlZjs7TLH2FfpS8DwW3_w/edit#heading=h.htn78a1hqoq)
```
[the remark this answers is not in the document]
Comment (SammySteiner) [reply documentation/decisions/2023-08-21-branch-conv-release-workflow.md:130]: never mind, I see you're addressing this in https://github.com/HHS/grants-equity/pull/431

File: documentation/decisions/2023-08-21-branch-conv-release-workflow.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+[Trunk Based Development](https://trunkbaseddevelopment.com/) is source-control branching model, where developers collaborate on code in a single branch called `main`.
+
+- **Pros**
+  - Widely adopted in general and preferred by the supporting engineering organization
+  - Facilitates continuos integration 
+  - Facilitates continuous delivery
+  - Facilitates lean experiments
+  - Simpler and more efficient than some other models
+- **Cons**
+  - Doesn't support releasing multiple versions of software, which is necessary in some cases (but not envisioned on this project)
```
[the remark this answers is not in the document]
Comment (acouch) [reply documentation/decisions/2023-08-21-branch-conv-release-workflow.md]: I updated this, thanks.

File: documentation/decisions/2023-08-21-branch-conv-release-workflow.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+The pros and cons for this strategy are described in reverse above. There is a higher congitive load for developers as each commit needs to be properly formatted, descriptive, and ideally atomic. However the history is more traversable if individual and tightly scoped commits are maintained.
+
+### Git commit conventions
+
+This encompasses the commit messages pushed to branches. This is tied to the decision to the branch merge strategy and assumes the use of the "squash and merge" strategy and github feature. The following options were considered:
+
+#### No commit convention
+
+- **Pros**
+  - Developers don't need to worry about the format of a commit while working on features
```
[the remark this answers is not in the document]
Comment (acouch) [reply documentation/decisions/2023-08-21-branch-conv-release-workflow.md]: I added those, with the caveat that it lowers overhead for some contributors (it doesn't for me personally), and that there are less to read through if the commits are shorter. Thanks!

**Pull request merged**